File: runinator-command-center/src/core/domain/models/gate/gate-record.ts
import type { JsonRecord, JsonValue } from "../../json";
import type { GateKind } from "./gate-kind";

export interface GateRecord {
  id?: string | null;
  workflow_run_id: string;
  node_id: string;
  kind: GateKind;
  status: string;
  label?: string | null;
  condition?: JsonValue;
  reason?: string | null;
  resolved_by?: string | null;
  resolved_at?: string | null;
  metadata?: JsonRecord;
  created_at?: string | null;
  updated_at?: string | null;
}

File: runinator-models/src/workflow_coordination/gate_record.rs
#[allow(unused_imports)]
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GateRecord {
    pub workflow_run_id: Uuid,
    pub node_id: String,
    pub kind: GateKind,
    pub status: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub label: Option<String>,
    #[serde(default)]
    pub condition: Value,
    pub metadata: Value,
}
